osbuild/buildroot.py

```diff
@@ -257,9 +260,40 @@ def run(self, argv, monitor, stage_timeout=None, binds=None, readonly_binds=None
             api_path = "/run/osbuild/api/" + api.endpoint
             mounts += ["--bind", api.socket_address, api_path]
 
+        index = osbuild.meta.Index(self._libdir)
+        def parse_manifest(path):
+            if path == "-":
+                manifest = json.load(sys.stdin)
+            else:
+                with open(path) as f:
+                    manifest = json.load(f)
+
+            return manifest
+        
+        desc = parse_manifest("test/data/manifests/fedora-ostree-image.json")
+        # detect the format from the manifest description
+        info = index.detect_format_info(desc)
+        if not info:
+            print("Unsupported manifest format")
+        fmt = info.module
+
+        # first thing is validation of the manifest
+        res = fmt.validate(desc, index)
+        if not res:
+            print("Error here.")
+
+        for type, name, version in index._schemata:
+            if type == 'Stage':
+                one = index._schemata.get((type, name, version))
+                print(one.name)
+                print(one.data['properties'])
```

Comment:
So at this point I have found the properties for a given stage,

Its time to add a caps field or something